kv/kvserver: TestFlowControlAdmissionPostSplitMergeV2 failed

kv/kvserver.TestFlowControlAdmissionPostSplitMergeV2 [failed](https://tanzanite.cluster.engflow.com/invocations/default/d3240df7-839f-42cb-8926-8d00294933c1?testReportRun=1&testReportShard=14&testReportAttempt=1#targets-Ly9wa2cva3Yva3ZzZXJ2ZXI6a3ZzZXJ2ZXJfdGVzdA==) on master @ [b38c8dd1faa68970c3310c1e82da71d593a67052](https://github.com/cockroachdb/cockroach/commits/b38c8dd1faa68970c3310c1e82da71d593a67052):


```
=== RUN   TestFlowControlAdmissionPostSplitMergeV2
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestFlowControlAdmissionPostSplitMergeV24052089275
    test_log_scope.go:81: use -show-logs to present logs inline
    flow_control_integration_test.go:2590: -- test log scope end --
test logs left over in: outputs.zip/logTestFlowControlAdmissionPostSplitMergeV24052089275
--- FAIL: TestFlowControlAdmissionPostSplitMergeV2 (232.22s)
=== RUN   TestFlowControlAdmissionPostSplitMergeV2/v2_enabled_when_leader_level=2
    flow_control_integration_test.go:4620: condition failed to evaluate within 3m45s: handle for 71 not found
    --- FAIL: TestFlowControlAdmissionPostSplitMergeV2/v2_enabled_when_leader_level=2 (228.21s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>deadlock=true</code>
 - <code>run=1</code>
 - <code>shard=14</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/kv
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestFlowControlAdmissionPostSplitMergeV2.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>


Jira issue: CRDB-42673

Epic CRDB-37515

--- a/pkg/kv/kvserver/flow_control_integration_test.go
+++ b/pkg/kv/kvserver/flow_control_integration_test.go
@@ -22,7 +22,6 @@ import (
 	"github.com/cockroachdb/cockroach/pkg/kv/kvserver"
 	"github.com/cockroachdb/cockroach/pkg/kv/kvserver/kvflowcontrol"
 	"github.com/cockroachdb/cockroach/pkg/kv/kvserver/kvflowcontrol/kvflowinspectpb"
-	"github.com/cockroachdb/cockroach/pkg/kv/kvserver/kvserverpb"
 	"github.com/cockroachdb/cockroach/pkg/roachpb"
 	"github.com/cockroachdb/cockroach/pkg/server"
 	"github.com/cockroachdb/cockroach/pkg/settings/cluster"
@@ -4798,11 +4797,7 @@ func (h *flowControlTestHelper) enableVerboseRaftMsgLoggingForRange(rangeID roac
 			&unreliableRaftHandler{
 				rangeID:                    rangeID,
 				IncomingRaftMessageHandler: si,
-				unreliableRaftHandlerFuncs: unreliableRaftHandlerFuncs{
-					dropReq: func(req *kvserverpb.RaftMessageRequest) bool {
-						return false
-					},
-				},
+				unreliableRaftHandlerFuncs: noopRaftHandlerFuncs(),
 			})
 	}
 }

